app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class Api::V1::UsersController < Api::V1::BaseController
+  before_action :set_user, only: [:show, :update]
+  before_action :authenticate_user!
+  before_action :check_access, only: [:update]
+
+  def index
+    @users = paginate(User.all)
```

Comment:
This will expose personally identifiable data including email to public

Reply:
Got it

Reply:
This will crash system as there are too many records in the system, add pagination

Reply:
The response is already paginated, can u please recheck once! @tachyons